### PR TITLE
blitz-auth & blitz-rpc: Fix unexpected infinite loops in codesandbox

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: https://github.com/blitz-js/blitz/issues/4191

### What are the changes and their implications?

Clears `HEADER_SESSION_CREATED` header so make sure it is set only when intended. In local testing (happens randomly only in dev env and codesandbox and its kind) it seems to fix the issue. Going to publish a temporary release to check in codesandbox.

Okay found the actual issue:

Screenshot 2023-08-17 at 13 49 07

## Bug Checklist

- [ ] Changeset added (run `pnpm changeset` in the root directory)
- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

### CODE REVIEW

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 1]
```diff
@@ -151,9 +151,11 @@ export async function getSession(
   res: ServerResponse,
   appDir = false,
 ): Promise<SessionContext> {
+  // clear any existing session-create headers
```
Comment (siddhsuresh) [inline packages/blitz-auth/src/server/auth-sessions.ts:154]:
```suggestion
```

**Comment by siddhsuresh** [comment]
!preview fix-infinite-loop

**Comment by WaysToGo** [comment]
@siddhsuresh I am getting the same infinite loop issue in the docker container deployed in AWS. getCurrentUser is called continuously.

**Comment by siddhsuresh** [comment]
@WaysToGo does this fix work for you?

can you minimally reproduce the issue ? 

Or maybe a video or screen shot with the network tab of the request or repo? Could also help in understanding the issue

**Comment by WaysToGo** [comment]
https://github.com/user-attachments/assets/607064db-2c97-4b6c-9346-c507250cc185

**Comment by WaysToGo** [comment]
https://github.com/user-attachments/assets/a70f70a5-2a05-4ef7-af52-2208cd72e728

it seems like everytime new cookie is been set and triggers reload.

**Comment by WaysToGo** [comment]
@siddhsuresh any update on this?

**Pull request closed without merging**